### PR TITLE
build: add nodejs/node update authors integration

### PR DESCRIPTION
**Pull request by bl0cknumber**
Add update authors script according to our latest license changes.

### CODE REVIEW

File: .github/update-authors.js
Comment (WikiRik) [inline .github/update-authors.js:33]: This wasn't very successful it seems. The following are still in the list;
renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Renovate Bot <bot@renovateapp.com>

**Review by WikiRik — COMMENTED** [review]
Don't forget to add the Action for it; https://github.com/nodejs/node/blob/master/.github/workflows/authors.yml

File: .github/update-authors.js
Comment (ephys) [reply .github/update-authors.js:33]: that's the private email feature from github :)

Comment (fncolon) [reply .github/update-authors.js:33]: > that's the private email feature from github :)

Yeahh i know

**Comment by fncolon** [comment]
> Could you add this file to the list of linted files in the eslint config?

Yup will add it on the weekend

File: .github/update-authors.js
Comment (fncolon) [reply .github/update-authors.js:33]: Okay i will using regex for `renovate[bot]@users.noreply.github.com` so we will make sure not to hardcoded every random numbers like `29139614+`

**Comment by fncolon** [comment]
> Could you add this file to the list of linted files in the eslint config?



> Could you add this file to the list of linted files in the eslint config?

Can you elaborate where the line I must add this? because i see there is line with `files: ['**/*.js'],`

**Review by ephys — CHANGES_REQUESTED** [review]
> Could you add this file to the list of linted files in the eslint config?

Turns out eslint ignores dotfolders by default.
If we want to lint .github, we can replace `ignorePatterns` with this (in `.eslintrc`):

```js
  ignorePatterns: [
    'dist/**/*',
    'types/**/*',
    'dev/**/*',
    '!.github',
  ],
```

However I think we should consider everything in `.github` to be reserved to github (as in they dictate what can be in it) and move the script to `/dev/update-authors`. Then change `ignorePatterns` to this:

```js
  ignorePatterns: [
    'dist/**/*',
    'types/**/*',
    'dev/**/*', // dev was temporarily ignored to ease transition to new linter
    '!dev/update-authors.js',
  ],
```

---

Also I see markdown files have all been changed. Can you revert these changes for now? So we can discuss them in a separate PR

**Comment by fncolon** [comment]
> > Could you add this file to the list of linted files in the eslint config?
> 
> Turns out eslint ignores dotfolders by default. If we want to lint .github, we can replace `ignorePatterns` with this (in `.eslintrc`):
> 
> ```js
>   ignorePatterns: [
>     'dist/**/*',
>     'types/**/*',
>     'dev/**/*',
>     '!.github',
>   ],
> ```
> 
> However I think we should consider everything in `.github` to be reserved to github (as in they dictate what can be in it) and move the script to `/dev/update-authors.js`. Then change `ignorePatterns` to this:
> 
> ```js
>   ignorePatterns: [
>     'dist/**/*',
>     'types/**/*',
>     'dev/**/*', // dev was temporarily ignored to ease transition to new linter
>     '!dev/update-authors.js',
>   ],
> ```
> 
> Also I see markdown files have all been changed. Can you revert these changes for now? So we can discuss them in a separate PR

Done, need your review

**Comment by fncolon** [comment]
Oops the lint worked and there's some errors i need to fix!

**Comment by fncolon** [comment]
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-regexp-test.md

Currently i don't know what the advantages using test over match, but here you are, I fixed it for the linter 😄 .

**Comment by ephys** [comment]
@fncolon Small perf benefit - `match` produces a throwaway array, `test` only returns a boolean :)

There are actually more lint issues but [I accidentally left `--fix` in `lint-no-fix`](https://github.com/sequelize/sequelize/pull/13930#discussion_r783512274) (the irony). We'll get it fixed shortly

**Comment by fncolon** [comment]
> @fncolon Small perf benefit - `match` produces a throwaway array, `test` only returns a boolean :)
> 
> There are actually more lint issues but [I accidentally left `--fix` in `lint-no-fix`](https://github.com/sequelize/sequelize/pull/13930#discussion_r783512274) (the irony). We'll get it fixed shortly

Thanks for the information

File: dev/update-authors.js
Comment (WikiRik) [inline dev/update-authors.js:64]:
```suggestion
  output.end("\n# Generated by dev/update-authors.js\n");
```

File: dev/update-authors.js
Comment (WikiRik) [inline dev/update-authors.js:4]:
```suggestion
// Usage: dev/update-author.js [--dry]
```

**Review by WikiRik — APPROVED** [review]
Two small suggestions, but apart from that I think we're good to merge this

File: .github/workflows/authors.yml
Comment (WikiRik) [inline .github/workflows/authors.yml:23]: Woops, did not notice this one before. I'll check if we have a GH_USER_TOKEN and if so which user is associated with that

Comment (WikiRik) [reply .github/workflows/authors.yml:23]: There is none currently, so we need to make a decision which account we want to use. Haven't encountered a sequelize bot account before, but might be nice to make for this and for the website deployments.
 @fncolon would you be okay if we wait a few days before merging this so @sdepold can look into this when he feels better again and has the time?

Comment (ephys) [reply .github/workflows/authors.yml:23]: It could use the GITHUB_TOKEN secret instead https://docs.github.com/en/actions/security-guides/automatic-token-authentication
It's a token for a bot managed by github. I've used it in the past to automatically comment coverage
image

Comment (WikiRik) [reply .github/workflows/authors.yml:23]: That would be even better!

Comment (sdepold) [reply .github/workflows/authors.yml:23]: 👍👍 

alternatively there is my personal GH_TOKEN

Comment (fncolon) [reply .github/workflows/authors.yml:23]: Woww that's so many option, but i implement the `GITHUB_TOKEN` thanks for the team work!

**Comment by fncolon** [comment]
Oops! some typo, i didn't rest well so lost focus, let me fix it today sun!

**Comment by bl0cknumber** [comment]
I think there's something wrong from the eslint script, so i changes that from this commit  https://github.com/sequelize/sequelize/commit/91ada1c56ba4ea33fdfd17a8fca0e30a17c11c09 from @ephys

**Comment by ephys** [comment]
there was a syntax error in it's config after your merge, just don't forget to re-update it's ignorePatterns

```
  ignorePatterns: [
    'dist/**/*',
    'types/**/*',
    'dev/**/*', // dev was temporarily ignored to ease transition to new linter
    '!dev/update-authors.js',
  ],
```

File: .github/workflows/authors.yml
Comment (sdepold) [inline .github/workflows/authors.yml:23]: I think we should use a different email address here. Feel free to use mail@sequelizejs.com

File: .eslintrc.js
Comment (sdepold) [inline .eslintrc.js:187]: Sorry if this is late to the party but should we maybe migrate this file to ts?

Comment (bl0cknumber) [reply .eslintrc.js:187]: > Sorry if this is late to the party but should we maybe migrate this file to ts?

I think it's feasible but would be overengineered, we need typescript for definition check only

Comment (bl0cknumber) [reply .eslintrc.js:187]: And update-authors scripts should be not touched by sequelize user in their app, so

Comment (bl0cknumber) [reply .eslintrc.js:187]: I think we don't need it to migrated to ts

File: .github/workflows/authors.yml
Comment (bl0cknumber) [reply .github/workflows/authors.yml]: Cool, Thanks!

File: .eslintrc.js
Comment (WikiRik) [reply .eslintrc.js:187]: Is this file the .eslintrc file? Or the authors script?

**Review by WikiRik — COMMENTED** [review]
Almost there

File: dev/update-authors.js
Comment (bl0cknumber) [reply dev/update-authors.js:2]: > Maybe use a permalink?

Can you elaborate about permalink?

File: .github/workflows/authors.yml
Comment (bl0cknumber) [reply .github/workflows/authors.yml:23]: Just following the sascha suggestion but maybe it's fine with the author even we don't need it, just for extra information.

File: dev/update-authors.js
Comment (WikiRik) [reply dev/update-authors.js:2]: Use this link so we know which 'version' of the file we used; https://github.com/nodejs/node/blob/c18ca140a12b543a3f970ef379f384ebd3297c3d/tools/update-authors.js

**Review by WikiRik — DISMISSED** [review]
Let's try this out with the author set

**Comment by ephys** [comment]
Last minute thought but I notice a lot of duplication in this file due to users changing their emails. Could we use the github API to unify them by github account? So

```
User Name <work-email>
User Name <home-email>
```

becomes

```
User Name <https://github.com/gh-username/>
```

**Comment by bl0cknumber** [comment]
> Last minute thought but I notice a lot of duplication in this file due to users changing their emails. Could we use the github API to unify them by github account? So
> 
> ```
> User Name <work-email>
> User Name <home-email>
> ```
> 
> becomes
> 
> ```
> User Name <https://github.com/gh-username/>
> ```

Look like this is an unresolved issue from node.js too, will try to fix it in this weekend!

https://github.com/nodejs/node/pull/11533

You can help if you want, but typings will asking for your help haha

**Comment by WikiRik** [comment]
> Last minute thought but I notice a lot of duplication in this file due to users changing their emails. Could we use the github API to unify them by github account? So
> 
> ```
> User Name <work-email>
> User Name <home-email>
> ```
> 
> becomes
> 
> ```
> User Name <https://github.com/gh-username/>
> ```

Not sure how easy that will become though. The email addresses used for the commits is what we use here, and those don't have to be registered with GitHub. I thought about it as well, but I couldn't think of a proper solution

**Comment by ephys** [comment]
True, maybe we can dedupe those that have an associated profile, and keep the email for those that do not

File: dev/update-authors.js
Comment (WikiRik) [inline dev/update-authors.js:34]: With the dependabot security update we had yesterday you need to exclude some additional emails;
<support@github.com> and <49699333+dependabot[bot]@users.noreply.github.com>
But maybe it would be better to generate the AUTHORS file based on the contributors' GitHub accounts ( https://github.com/sequelize/sequelize/graphs/contributors ) which I've seen in some other projects as well

Comment (bl0cknumber) [reply dev/update-authors.js:34]: @WikiRik things to note

From https://docs.github.com/en/repositories/viewing-activity-and-data-for-your-repository/viewing-a-projects-contributors
If you don't appear in a repository's contributors graph, it may be because:
- You aren't one of the top 100 contributors.
- Your commits haven't been merged into the default branch.
- The email address you used to author the commits isn't connected to your account on GitHub.

Comment (bl0cknumber) [reply dev/update-authors.js:34]: adding filter is preferred than losing another contributors data if the contributors is more than 100.

Comment (WikiRik) [reply dev/update-authors.js:34]: Ah, then it's not better to generate it based on that.
Did you think about a way of combining duplicate emails if they are both associated with a GitHub account?

Comment (sdepold) [reply dev/update-authors.js:34]: If I may add my 2 cents: I don't think duplication is a problem 😊🙈

Comment (bl0cknumber) [reply dev/update-authors.js:34]: > If I may add my 2 cents: I don't think duplication is a problem 😊🙈

Since there's no prohibition against duplicate emails, even if it's better because it's like nodes showing changes to authors emails, at least we've filtered emails for bots. Maybe this is enough to merge. Any thoughts on this? @ephys @WikiRik I will change my habit to rebase

Comment (bl0cknumber) [reply dev/update-authors.js:34]: > If I may add my 2 cents: I don't think duplication is a problem 😊🙈

Since there's no prohibition against duplicate emails, even if it's better because it's like nodes showing changes to authors emails, at least we've filtered emails for bots. Maybe this is enough to merge. Any thoughts on this? @ephys @WikiRik

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
@ephys you requested some changes last month, are they still valid or have they been handled now?

**Comment by ephys** [comment]
No I think we're good to go
I assume it won't try to open a PR if there are no new changes :)

**Review by ephys — APPROVED** [review]

**Comment by bl0cknumber** [comment]
I think @sdepold should approve this PR too after looks up to his review, maybe we can create another related PR if I missing something, I will merge this :)

File: AUTHORS
Comment (sdepold) [inline AUTHORS:10]: Gosh. You know how old I've been when I worked with Meg?

Comment (sdepold) [reply AUTHORS:10]: @megshark nostalgic 👋  from me

Same to @mickhansen and @janmeier

**Comment by sdepold** [comment]
Love it. Kudos to the peeps!

File: AUTHORS
Comment (bl0cknumber) [reply AUTHORS:10]: @sdepold I'm almost in tears seeing what progress you've start so far, thank you for your hard work!

**Comment by megshark** [comment]
Nice thanks!!! Please update my email to ***@***.*** blushing!

Meg

**Comment by WikiRik** [comment]
@megshark The email addresses are based on the ones provided with the merged commits, but maybe @bl0cknumber can look into the option to change email addresses. You would however need to post your updated email in a comment through the GitHub website since GitHub redacts email addresses sent over email replies.

**Comment by megshark** [comment]
OK sure, thanks!  Metamarkets was acquired by Snapchat in 2017 (leaving an
open source legacy of Apache Druid).  My github is megshark at Gmail but I
can try as you suggest.

I miss NodeJS and Sequelize!  Back in the Java world for now... ;)

-Meg

File: AUTHORS
Comment (janmeier) [reply AUTHORS:10]: Yeah, good times, good times! I miss it :wave:

**Comment by WikiRik** [comment]
@fzn0x I just saw the failed runs for this, can you check it out? Seems to be a permission issue

**Comment by fzn0x** [comment]
> @fzn0x I just saw the failed runs for this, can you check it out? Seems to be a permission issue

Let me see if I can fix it

**Comment by WikiRik** [comment]
Should just be `chmod +x dev/update-authors.js` and re-committing the file

**Pull request merged**
